Leetcode/Python/89_Gray_Code.py

```diff
@@ -0,0 +1,25 @@
+"""
+89. Gray Code
+An n bit gray code is a sequence of binary numbers differing by a single bit
+It has 2**n entries starting from 0.
+The first and the last entry also differ by a single bit
+
+Difficulty:- Medium
+
+https://leetcode.com/problems/gray-code/
+Input: n = 2
+Output: [0, 1, 3, 2]
+
+Necessary theory:-
+^ = Bitwise XOR operator
+>> = Bitwise right shift operator
+author:- aarifsk
+"""
+
+
+class Solution:
+    def gray_code(self, n: int) -> List[int]:
+        answer = []
+        for idx in range(2**n):
+            answer.append(idx ^ (idx >> 1))
```

Comment:
How do this XOR and Right Shift work here?
for example 00 01 10 11 as idx is 0, 1, 2 and 3.